pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -132,6 +133,12 @@ func (h *upgradeHandler) OnChanged(key string, upgrade *harvesterv1.Upgrade) (*h
 		return h.upgradeClient.Update(toUpdate)
 	}
 
+	// only run further operations for latest upgrade
+	if upgrade.Labels == nil || upgrade.Labels[harvesterLatestUpgradeLabel] != "true" {
+		return upgrade, nil
+	}
+
+	logrus.Infof("handle upgrade %s/%s", upgrade.Namespace, upgrade.Name)
```

Comment:
Is it possible to move this condition checking before L91? If I understand it correctly, we only want to handle the upgrade CR which is the "latest", we can short-circuit others in the early stage.

Reply:
We only want to handle latest upgrade CR. However, when users create a new upgrade, there is no `harvesterhci.io/latestUpgrade` on it, so we need L91 to initialize the label. Thanks.

Reply:
Thanks for the explanation!